src/api/c/DependencyModule.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
```

Comment:
May be more appropriate location for the `DependencyModule` header and source file is `src/backend/common` where the code common to all backends is present. The classes and functions from that location are used both at `src/api/c/` level and `src/backend/[cpu | cuda | opencl ]` level. It looks like there is a very good change that we might end up using this DependencyModule class at backend level also for other libraries - may be when you do similar changes to `forge` itself.

Reply:
Addressed.